python/test/t_SmolyakExperiment_std.py

```diff
@@ -0,0 +1,309 @@
+    sortedWeights = weights[indices]
+    return sortedNodes, sortedWeights
+
+
+def printNodesAndWeights(nodes, weights):
+    """
+    Print nodes and weights of an Experiment.
+    Parameters
+    ----------
+    nodes : ot.Sample(size, dimension)
+        The sorted nodes.
+    weights : ot.Point(size)
+        The weights.
+    """
+    size = nodes.getSize()
+    if weights.getDimension() != size:
+        raise ValueError(
+            "Nodes have size %d, but weights have dimension %d"
+            % (size, weights.getDimension())
+        )
+    dimension = nodes.getDimension()
+    for i in range(size):
+        row = "[%d] %.4f : " % (i, weights[i])
+        for j in range(dimension):
+            row += "%.4f, " % (nodes[i, j])
+        print(row)
+    return
+
+
+def printNodes(nodes):
+    """
+    Print nodes of an Experiment.
+    Parameters
+    ----------
+    nodes : ot.Sample(size, dimension)
+        The sorted nodes.
+    """
+    size = nodes.getSize()
+    dimension = nodes.getDimension()
+    for i in range(size):
+        row = "[%d] : " % (i)
+        for j in range(dimension):
+            row += "%.4f, " % (nodes[i, j])
+        print(row)
+    return
+
+
+def roundSample(sample, numberOfDigits):
+    """
+    Round a sample
+
+    Parameters
+    ----------
+    sample : ot.Sample(size, dimension)
+        The sample.
+    numberOfDigits : int
+        The number of decimal digits to keep.
+
+    Returns
+    -------
+    roundedSample : ot.Sample(size, dimension)
+        The rounded sample.
+    """
+    size = sample.getSize()
+    dimension = sample.getDimension()
+    factor = 10.0**numberOfDigits
+    for i in range(size):
+        for j in range(dimension):
+            rounded = (int)(factor * sample[i, j])
+            sample[i, j] = rounded / factor
+    return sample
+
+
+def testSmolyakExperiment1():
+    # Generate a Smolyak Gauss-Legendre rule in 2 dimensions.
+    print("testSmolyakExperiment1:")
+    experiment1 = ot.GaussProductExperiment(ot.Uniform(0.0, 1.0))
+    experiment2 = ot.GaussProductExperiment(ot.Uniform(0.0, 1.0))
+    collection = [experiment1, experiment2]
+    level = 3
+    smolyak = otexp.SmolyakExperiment(collection, level)
+    nodes, weights = smolyak.generateWithWeights()
+    numberOfDigits = 14
+    nodes = roundSample(nodes, numberOfDigits)
+    nodes, weights = sortNodesAndWeights(nodes, weights)
+    print("Computed:")
+    printNodesAndWeights(nodes, weights)
+    # Sort
+    nodesExact = ot.Sample(
+        [
+            [0.112702, 0.5],
+            [0.211325, 0.211325],
+            [0.211325, 0.5],
+            [0.211325, 0.788675],
+            [0.5, 0.112702],
+            [0.5, 0.211325],
+            [0.5, 0.5],
+            [0.5, 0.788675],
+            [0.5, 0.887298],
+            [0.788675, 0.211325],
+            [0.788675, 0.5],
+            [0.788675, 0.788675],
+            [0.887298, 0.5],
+        ]
+    )
+    weightsExact = ot.Point(
```

Comment:
If the reference values are given in the test script anyway, you can easily convert all print-based tests to assert-based ones.